examples/camera-app/linux/src/clusters/webrtc-provider/webrtc-provider-manager.cpp

```diff
@@ -609,72 +609,76 @@ void WebRTCProviderManager::OnDeviceConnected(void * context, Messaging::Exchang
     WebRTCProviderManager * self = reinterpret_cast<WebRTCProviderManager *>(context);
     VerifyOrReturn(self != nullptr, ChipLogError(Camera, "OnDeviceConnected:: context is null"));
 
-    WebrtcTransport * transport = nullptr;
-    uint16_t sessionId          = 0;
+    // Derive sessionId from sessionHandle by looking up the peer node ID
+    NodeId peerNodeId = sessionHandle->GetPeer().GetNodeId();
+    auto sessionIt    = self->mSessionIdMap.find(peerNodeId);
```

Comment:
A peer node ID is not a unique identifier, right?  There can be multiple peers, on different fabrics, with the same node ID.